sys/kern/src/kipc.rs

```diff
@@ -462,7 +463,9 @@ fn software_irq(
         )))?;
 
     for &irq in irqs.iter() {
-        crate::arch::pend_software_irq(irq);
+        // Any error here would be a problem in our dispatch table, not the
+        // caller, so we panic because we want to hear about it.
+        crate::arch::pend_software_irq(irq).unwrap_lite();
```

Comment:
Hmm --- this is the only call site of `pend_software_irq`, right? So, if we're going to unwrap here, we _could_ just leave that function using infallible array indexing and let it panic, instead. I dunno if that has a meaningful impact on binary size, though, and this is maybe nicer in the event that we add other uses of `pend_software_irq` later, which _don't_ want to panic. I think having it be the caller's decision whether or not to panic is almost certainly the ideologically nicer factoring...